# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientManager.h
# Commit: 6125bb02952a
# Full Hash: 6125bb02952aff71c0d757e238a7752101b2d6c2
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-09-03 04:06:20
# Regressor Bug: 1556489
# File Overlap Count: 1
# Description:
#   Bug 1556489 - P23 - Only create new ClientSource objects in the content, but use the recreated info from redirects in the parent. r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D42260
# ==============================================================================

diff -r 06529d12b589 -r 6125bb02952a dom/clients/manager/ClientManager.h
--- a/dom/clients/manager/ClientManager.h	Mon Aug 26 12:18:57 2019 +1000
+++ b/dom/clients/manager/ClientManager.h	Tue Aug 27 08:53:37 2019 +1000
@@ -52,6 +52,9 @@
       ClientType aType, nsISerialEventTarget* aEventTarget,
       const mozilla::ipc::PrincipalInfo& aPrincipal);
 
+  UniquePtr<ClientSource> CreateSourceInternal(
+      const ClientInfo& aClientInfo, nsISerialEventTarget* aEventTarget);
+
   already_AddRefed<ClientHandle> CreateHandleInternal(
       const ClientInfo& aClientInfo, nsISerialEventTarget* aSerialEventTarget);
 
@@ -84,6 +87,17 @@
       ClientType aType, nsISerialEventTarget* aEventTarget,
       const mozilla::ipc::PrincipalInfo& aPrincipal);
 
+  // Construct a new ClientSource from an existing ClientInfo (and id) rather
+  // than allocating a new id.
+  static UniquePtr<ClientSource> CreateSourceFromInfo(
+      const ClientInfo& aClientInfo, nsISerialEventTarget* aSerialEventTarget);
+
+  // Allocate a new ClientInfo and id without creating a ClientSource. Used
+  // when we have a redirect that isn't exposed to the process that owns
+  // the global/ClientSource.
+  static Maybe<ClientInfo> CreateInfo(ClientType aType,
+                                      nsIPrincipal* aPrincipal);
+
   static already_AddRefed<ClientHandle> CreateHandle(
       const ClientInfo& aClientInfo, nsISerialEventTarget* aSerialEventTarget);
 